contracts/ERC721A.sol

```diff
@@ -809,11 +859,22 @@ contract ERC721A is IERC721A {
      * Emits a {ConsecutiveTransfer} event.
      */
     function _mintERC2309(address to, uint256 quantity) internal virtual {
-        uint256 startTokenId = _currentIndex;
-        if (to == address(0)) revert MintToZeroAddress();
-        if (quantity == 0) revert MintZeroQuantity();
-        if (quantity > _MAX_MINT_ERC2309_QUANTITY_LIMIT) revert MintERC2309QuantityExceedsLimit();
+        assembly {
+            if eq(to, 0) {
```

Comment:
not safe. Upper bits of to may not be clean.

Reply:
can you elaborate a bit, since when 'to' is passed as an address, it should be a 160 bits value. and the remaining bits supposedly should be set to 0 value.

Reply:
The upper 96 bits may not be cleaned, depending where `to` comes from.

When using regular Solidity, the compiler automatically cleans it lazily when writing to memory, storage, or doing comparisons. No guarantees in assembly.

Reply:
ohhh i see, thanks for the clarification :)